base_layer/wallet/src/utxo_scanner_service/utxo_scanning.rs

```diff
@@ -584,14 +589,16 @@ where TBackend: WalletBackend + 'static
                         return Ok(());
                     },
                     Err(e) => {
+                        let wait_seconds = 30;
                         warn!(
                             target: LOG_TARGET,
-                            "Failed to scan UTXO's from base node {}: {}", peer, e
+                            "Failed to scan UTXO's from base node {} ({}), waiting {}s", peer, e, wait_seconds
                         );
                         self.publish_event(UtxoScannerEvent::ScanningRoundFailed {
                             num_retries: self.num_retries,
                             retry_limit: self.retry_limit,
                         });
+                        time::delay_for(Duration::from_secs(wait_seconds)).await;
```

Comment:
We should not put a wait here. Rather put the wait where this run function is called from or on in the none part. 

Basically what's going to happen here is that say the peer has an error the wallet will wait 30 seconds before trying the next peer. In UTXO recovery there is more than 1 peer so we want to hop to the next peer as fast as possible. 

I strongly prefer putting the wait in the where this task is called from. This to me is a dirty hack to fix a problem we are effectively only getting on cucumber, and only in the logs, where the wallet cant scan yet because the base_node it is connected to has not finished booting up.

Reply:
Yip, this delay is not good, removed it.